bootstrap-orbit.sh

```diff
@@ -6,26 +6,70 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
-echo "Installing required system dependencies..."
-sudo add-apt-repository universe
-if [ $? -ne 0 ]; then
-  sudo apt-get install -y software-properties-common
-  sudo add-apt-repository universe
-fi
+OPTIONS=$(getopt -o "" -l "force-public-remotes,assume-linux,assume-windows,ignore-system-requirements,dont-compile" -- "$@")
+eval set -- "$OPTIONS"
+
+CONFIG_INSTALL_OPTIONS=""
+IGNORE_SYS_REQUIREMENTS=""
+DONT_COMPILE=""
+
+while true; do
+  case "$1" in
+    --force-public-remotes|--assume-linux|--assume-windows)
+      CONFIG_INSTALL_OPTIONS="${CONFIG_INSTALL_OPTIONS} $1"; shift;;
+    --ignore-system-requirements) IGNORE_SYS_REQUIREMENTS="yes"; shift;;
+    --dont-compile) DONT_COMPILE="yes"; shift;;
+    --) shift; break;;
+  esac
+done
+
+readonly REQUIRED_PACKAGES=( build-essential libglu1-mesa-dev mesa-common-dev \
+                             libxmu-dev libxi-dev libopengl0 qt5-default \
+                             libxxf86vm-dev python3-pip )
+
+if [[ $IGNORE_SYS_REQUIREMENTS != "yes" ]]; then
+  if which dpkg-query >/dev/null; then
+    readonly installed="$(dpkg-query --show -f'${Package}\n')"
+    PACKAGES_MISSING="no"
+    for package in ${REQUIRED_PACKAGES[@]}; do
+      if ! egrep "^${package}$" <<<"$installed" > /dev/null; then
+        PACKAGES_MISSING="yes"
+        break
+      fi
+    done
+
+    if [[ $PACKAGES_MISSING == "yes" ]]; then
+      echo "Installing required system dependencies..."
 
-sudo apt-get update
-sudo apt-get install -y build-essential
-sudo apt-get install -y libglu1-mesa-dev mesa-common-dev libxmu-dev libxi-dev
-sudo apt-get install -y libopengl0
-sudo apt-get install -y qt5-default
-sudo apt-get install -y libxxf86vm-dev
-sudo apt-get install -y python3-pip
+      # That only works on Ubuntu!
+      if [[ "$(lsb_release -si)" == "Ubuntu" ]]; then
+        sudo add-apt-repository universe
+        if [ $? -ne 0 ]; then
+          sudo apt-get install -y software-properties-common
+          sudo add-apt-repository universe
+        fi
+      fi
+
+      sudo apt-get update || exit $?
+      sudo apt-get install -y ${REQUIRED_PACKAGES[@]} || exit $?
+    fi
```

Comment:
maybe extract this to a separate function?

Reply:
good idea. done.